packages/controller/main.js

```diff
@@ -695,6 +696,24 @@ function createObjects(onConnect) {
                             delete hostAdapter[id];
                         }
                     }
+                } else if (obj.type && obj.type === 'host' && obj.common && obj.common.installedVersion) {
+                    // host object changed
+                    // TODO: remove this shim if 4.0 is old enough
+                    if (controllerVersions[id]) {
+                        if (semver.lt(controllerVersions[id], '4.0.0') && semver.gte(obj.common.installedVersion, '4.0.0')) {
```

Comment:
Here we need a todo because as discussed should be adjusted

Reply:
what do you exactly mean, beside the todo in L701